### PR TITLE
cli: show timestamp in local timezone and without millis and offset

### LINKED ISSUE #2900: FR: standardize jj log output timezone
**Is your feature request related to a problem? Please describe.**
I just updated my clone of the jj upstream code, and was struck by the fact that my working directory is on top of a change in the "future" (because it came from Japan) - clearly, this is a bit confusing (and probably even more so when the timezone difference is smaller).

**Describe the solution you'd like**
In order to make times more easily comparable, it'd probably make sense to always output in the local timezone unless the user explicit asks it to do otherwise?

**Describe alternatives you've considered**
Always outputting in an arbitrary time zone (UTC?) would also make them comparable, but still harder to make sense of vs "now".

**Additional context**
```
@  pwrrtvzoxqop rdamazio@google.com 2024-01-30 20:47:45.000 -08:00 44b5a0c20f9f
│  (empty) (no description set)
◉  yyqlyqtqwvrz yuya@tcha.org 2024-01-31 09:40:52.000 +09:00 main 976b80120885
│  index: on reinit(), delete all segment files to save disk space
~
```

(also, what's with the fractional seconds? Are those ever relevant?)

### PR DESCRIPTION
**Pull request by martinvonz**
As discussed in #2900, the milliseconds are rarely useful, and it can be confusing with different timezones because it makes harder to compare timestamps.

Closes #2900

# Checklist

If applicable:
- [x] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [x] I have added tests to cover my changes

### CODE REVIEW

**Comment by martinvonz** [comment]
Does anyone know how to set the timezone on Windows for testing? I set `$TZ` in the test setup but Windows doesn't care.

**Comment by martinvonz** [comment]
This should be ready for review (hopefully tests pass now)

**Review by yuja — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -89,6 +89,9 @@ No code changes (fixing Rust `Cargo.toml` stuff).
 * In the templating language, Timestamps now have a `.local()` method for
   converting to the local timezone.
 
+* Timestamps are now shown in local timezone and without milliseconds and
+  timezone offset by default.
+
```
Comment (yuja) [inline CHANGELOG.md:94]: nit: move to the `Unreleased` section

Comment (martinvonz) [reply CHANGELOG.md]: Done.

**Pull request merged**